Implemented a right-click menu for notification icon.

Eiyeron opened the pull request:
Hello.

I quickly drafted a right-click menu for the notification icon to quickly have an exit option. I also threw an option to force errored cooking commands as I sometimes get to retrigger those after a few errors.
![image](https://github.com/user-attachments/assets/5162b371-2dc9-4130-93cb-6e476359cd4e)

This is more a draft than a serious PR but if you're interested into this PR, I could write a less hacky-looking solution than the one provided here.

Have a nice day.

Srekel commented:
> Cheers, I'll try to take a look tonight.
> 
> Out of curiosity, why do you need the cook errored button? Normally you'd have to change an input to fix an error, if you do that it should recook automatically. Is that not working as intended?

It's quite common for us to need to recook errored a few times to get them all successfully cooked.

jlaumon commented on src/Notifications.h:
```diff
@@ -7,6 +7,11 @@
 constexpr uint32 cNotifCallbackID = 0x8001; //  ie. WM_APP + 1
+// Not used yet but could be used as an ID mask to dispatch events.
+[[maybe_unused]]
+constexpr uint32 cNotifMenuId = 0x8100;
```
Does that value mean something specific? Or it's just a value that you picked?

Eiyeron replied:
I just used `0x81xx` as a pseudo-namespace where the least significant byte would be the notification menu's command ID. The upper byte would just identifying the context and masking would be easier to think about.

Just some forward thinking on the spot, not really useful as is *for now*.

jlaumon commented on src/main.cpp:
```diff
@@ -530,7 +552,26 @@ LRESULT WINAPI WndProc(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam)
+
+			HMENU hMenu = CreatePopupMenu();
+			gApp.mNotifMenuHmenu = hMenu;
+			ret = InsertMenu(hMenu, 0, MF_BYPOSITION | MF_STRING, cNotifMenuCookErrored, L"Cook errored");
```
In the rest of the code I've tried to use the A version of the Windows functions instead of the WCHAR ones. They should support receiving utf-8 strings thanks to the `enable_utf8.manifest` (but does need checking, because some of then don't work!)

Just try `InsertMenuA(..., u8"Cook errored 😭")` and see what happens :D

Eiyeron replied:
My bad, I noticed usage of W function in the same file so I went with the flow. I'm going to change them then, thanks for pointing at it.

Eiyeron commented:
As discussed on the Fediverse instead of here due to an issue accessing GitHub I also removed the test command and added a pause cooking button instead.

jlaumon commented:
Looks good to me! I'll give it a try tonight and merge if I see no issues 👍
